spec/agent/navigation_spec.rb

```diff
@@ -0,0 +1,85 @@
+describe AmaLayout::Agent::Navigation do
+  describe "#nav_file_path" do
+    let(:file_path) { File.join(Gem.loaded_specs["ama_layout"].full_gem_path, "lib", "ama_layout", "agent_navigation.yml") }
+
+    it "defaults to lib/ama_layout/agent_navigation.yml" do
+      expect(subject.nav_file_path).to eq file_path
+    end
+
+    context "overridden file path" do
+      let(:file_path) do
+        File.join(Gem.loaded_specs["ama_layout"].full_gem_path, "spec", "ama_layout", "fixtures", "agent_navigation.yml")
+      end
+      let(:user) { double("user") }
+      let(:subject) { described_class.new(user: user, nav_file_path: file_path, current_url: '/') }
+
+      it "uses the overridden file path" do
+        expect(subject.items.first.text).to eq "Waffles"
+      end
+    end
+  end
+
+  describe "#items" do
+    it "does not remove nil #navigation_items" do
+      expect(subject.items.collect {|i| i.alt }).to include nil
+    end
+
+    it "displays links " do
+      expect(subject.items.collect {|i| i.link }).to eq subject.navigation_items.collect {|i| i["link"] }
```

Comment:
Could use `.collect(&:link)`, but I don't think the `i["link"]` part has a shortcut like that.